File: examples/python/best-practices/dynamic-operation-names.py
@durable_execution
def lambda_handler(event: dict, context: DurableContext) -> list:
    results = []
    for i, item in enumerate(event["items"]):
        result = context.step(
            process_item(item),
            name=f"process_item_{i}_{item['id']}"
        )
        results.append(result)
    return results

File: examples/java/best-practices/dynamic-operation-names.java
var items = (List<Map<String, Object>>) input.get("items");
var results = new ArrayList<>();
for (int i = 0; i < items.size(); i++) {
    final var item = items.get(i);
    var result = ctx.step(
        "process_item_" + i + "_" + item.get("id"), Object.class,
        stepCtx -> processItem(item));
    results.add(result);
}
return results;
